src/select/__tests__/virtual-scroll.test.tsx

```diff
@@ -0,0 +1,62 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import { render } from '@testing-library/react';
+import Select, { SelectProps } from '../../../lib/components/select';
+import createWrapper from '../../../lib/components/test-utils/dom';
+
+const defaultProps: SelectProps = {
+  options: [
+    { value: '1', label: 'One' },
+    { value: '2', lang: 'Two' },
+    { value: '3', lang: 'Three' },
+  ],
+  virtualScroll: true,
+  selectedOption: null,
+  onChange: () => {},
+};
+
+function renderWithWrapper(ui: React.ReactElement) {
+  const { container } = render(ui);
+  return createWrapper(container).findSelect()!;
+}
+
+describe('Virtual scroll support', () => {
+  test('should render plain virtual list', () => {
+    const wrapper = renderWithWrapper(<Select {...defaultProps} />);
+    wrapper.openDropdown();
+    expect(wrapper.findDropdown().findOptions()).toHaveLength(3);
+  });
+
+  test('should render a subset of items in virtual list', () => {
+    const wrapper = renderWithWrapper(
+      <Select {...defaultProps} options={Array.from({ length: 100 }, (_, index) => ({ value: `Option ${index}` }))} />
+    );
+    wrapper.openDropdown();
+    expect(wrapper.findDropdown().findOptions()).toHaveLength(11);
```

Comment:
(same q here)

Reply:
For some reason, the overscan count doubles, I am guessing because it works two way due to some implementation details

Reply:
seems pretty strange and likely to cause future flakiness, but I'm happy if you're convinced it should be stable